component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthenticator.java

```diff
@@ -106,6 +110,12 @@ private void readServerConfig(){
     public boolean authenticateClient(HttpServletRequest httpServletRequest, Map<String, List> bodyParameters,
                                       OAuthClientAuthnContext oAuthClientAuthnContext) throws OAuthClientAuthnException {
 
+        if (Boolean.parseBoolean(isFapiApplication(getClientId(httpServletRequest, bodyParameters,
+                oAuthClientAuthnContext)))) {
+            if (!isRegisteredSignatureAlgorithm(httpServletRequest, bodyParameters, oAuthClientAuthnContext)) {
```

Comment:
Add bit inline comments..

Reply:
Fixed in c18c85cc171d19c53412ac6eaad0e1e4b738d909